recursive-methods.rb

```diff
@@ -1,34 +1,128 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
+
+# time: O(n) because it needs to execute n times to calculate factorial
+# space: O(n) because of the stack space (n calls to factorial)
 def factorial(n)
-  puts "Not implemented."
+  if n == 0 || n == 1
+    return 1
+  else
+    return n * (factorial n - 1)
```

Comment:
It's interesting that this works in Ruby: `factorial n - 1`. I would have thought it'd be a syntax error if it didn't look like this: `factorial(n-1)`